src/coreclr/vm/corelib.h

```diff
@@ -1040,6 +1040,21 @@ DEFINE_METHOD(HANDLE_MARSHALER,          CONVERT_SAFEHANDLE_TO_NATIVE,ConvertSaf
 DEFINE_METHOD(HANDLE_MARSHALER,          THROW_SAFEHANDLE_FIELD_CHANGED, ThrowSafeHandleFieldChanged, SM_RetVoid)
 DEFINE_METHOD(HANDLE_MARSHALER,          THROW_CRITICALHANDLE_FIELD_CHANGED, ThrowCriticalHandleFieldChanged, SM_RetVoid)
 
+#ifdef TARGET_WINDOWS
```

Comment:
I assume that this is two-liner because ILLINK chokes on `#if defined(TARGET_X86) && defined(TARGET_WINDOWS)`

Reply:
Sadly, yes.